src/app/pages/rpc-get-addresses/components/get-addresses.layout.tsx

```diff
@@ -1,63 +1,108 @@
-import { Box, Flex, styled } from 'leather-styles/jsx';
+import type { ReactNode } from 'react';
 
-import { Button, LeatherLettermarkIcon, LeatherLogomarkIcon } from '@leather.io/ui';
+import { motion, useAnimationControls } from 'framer-motion';
+import { Box, Flex, VStack, styled } from 'leather-styles/jsx';
 
-import { RequesterFlag } from '@app/components/requester-flag';
+import {
+  Approver,
+  Button,
+  CheckmarkIcon,
+  Favicon,
+  Flag,
+  LeatherLogomarkIcon,
+} from '@leather.io/ui';
+import { delay } from '@leather.io/utils';
+
+import { closeWindow } from '@shared/utils';
 
 interface GetAddressesLayoutProps {
   requester: string;
-  onUserApproveGetAddresses(): void;
+  switchAccount: ReactNode;
+  onUserApprovesGetAddresses(): void;
 }
 export function GetAddressesLayout({
   requester,
-  onUserApproveGetAddresses,
+  switchAccount,
+  onUserApprovesGetAddresses,
 }: GetAddressesLayoutProps) {
+  const originLogoMovesToCenter = useAnimationControls();
+  const contentDisappears = useAnimationControls();
+  const checkmarkEnters = useAnimationControls();
+
+  async function userApprovesGetAddresses() {
+    await contentDisappears.start({ opacity: 0, transition: { duration: 0.4 } });
+    await originLogoMovesToCenter.start({
+      transform: 'scale(1.4) translate(104px, 142px) rotate(-20deg)',
+      transformOrigin: 'center center',
+    });
+    await checkmarkEnters.start({ transform: 'scale(0.8)' });
+    await delay(350);
+    onUserApprovesGetAddresses();
+    await delay(1400);
+  }
+
   return (
-    <Flex
-      flexDirection="column"
-      height="100vh"
-      width="100%"
-      maxWidth="popupMaxWidth"
-      margin="0 auto"
-    >
-      <Flex
-        flex={1}
-        flexDirection="column"
-        textAlign="center"
-        alignItems="center"
-        p="space.06"
-        gap="space.06"
+    <Flex flexDir="column" background="ink.background-secondary" height="100vh">
```

Comment:
Maybe we could create more `.layout` wrapper files here as it's getting quite big. 

Could all the `motion.div` stuff work in a standalone file without drilling?

Reply:
What would be the individual responsibilities of the separate layout components?

The motion stuff is of course kinda verbose, but it's all part of the layout so wonder if separating further would make it harder to manage.

Reply:
It would be for readability only as the file is quite large. 

For example as we are passing so many props here it could be it's own component:
```
 <Box
          mx="space.05"
          mt="space.05"
          bg="ink.background-primary"
          p="space.04"
          borderRadius="lg"
          display="inline-block"
          borderColor="ink.border-default"
          borderWidth="1px"
          width="72px"
          height="72px"
        >
          <Favicon origin={requester} size={64} />
</Box>
```

Reply:
Made a component for this one